File: 202 Happy Number/solution.py
class Solution:
    # @param {integer} n
    # @return {boolean}
    def isHappy(self, n):
        resultset = set()
        while n!=1 and n not in resultset:
            resultset.add(n)
            happy = 0
            while n:
                m,digit = divmod(n,10)
                happy+=digit*digit
                n = m
            n = happy
        return n==1
        

File: 202 Happy Number/solution.cpp
class Solution {
public:
    bool isHappy(int n) {
       set<int> resultset;
       while (n!=1&&resultset.find(n)==resultset.end()){
           resultset.insert(n);
           int result = 0;
           while (n!=0){
               int digit = n%10;
               result+=digit*digit;
               n = n/10;
           }
           n = result;
       }
       return n==1;
    }
};